### DISCUSSION TITLE
trying to get printing new formulae/casks back on `brew update`

### DISCUSSION THREAD

**Discussion by marcotrosi** [Homebrew/discussions#3353] [Everyday usage] [answered]
Hi everyone,
I noticed that some days/weeks ago `brew update` stopped printing the list of new formulae/casks. So I did some investigations and I found out that it was removed in the last version.
Then I found a variable of which I thought it could bring back the list of new packages.
`export HOMEBREW_UPDATE_REPORT_ALL_FORMULAE=1`
Now it is printing lists of modified formulae and casks.
Looks like this ...
```
Updated Homebrew from a5a5d927f to ad13524bc.
Updated 2 taps (homebrew/core and homebrew/cask).
==> Modified Formulae
ansible        brigade-cli    cadence
==> Modified Casks
anaconda                   deezer
```
Here are my questions ...

- does this list of modified packages also contain the new packages?
- or is it pure coincidence that I had no new packages so far, and they would be listed here if any would exist?
- or was printing new packages removed completely? (oh please say no)
- is there maybe another variable or flag that can turn on printing new packages?

Personally I would also be interested in a command to print new packages. For example "print the 50 latest packages added" or "print all new packages added since 'date'|'days ago'". Because so far I was only able to see the new packages when running `brew update`, and after that I found no way to print them again. So when I didn't the have the time to check the new packages on an update, then there was no way to check them later unless I would have saved the output.

thank you

marco

**Comment by SMillerDev** [comment, marked as answer]
That variable restores the old behavior so that includes new formula.

**Reply by marcotrosi** [reply]
> That variable restores the old behaviour so that includes new formula.

ah great, thank you @SMillerDev 
and are the new packages within the sections where it says "modified" or didn't I have by coincidence any new packages and they will be listed separately?

p.s. the brew update output prints "formulae" and you write "formulas"
is "formulae" a typo as the letter e is close to s?

**Reply by Bo98** [reply]
There is a bug where new formulae is not being showed, but we're working on fixing that: https://github.com/Homebrew/brew/pull/13403

> is "formulae" a typo as the letter e is close to s?

Both "formulae" and "formulas" are recognised plural forms of "formula". We generally use the former.

**Reply by marcotrosi** [reply]
> There is a bug where new formulae is not being showed, but we're working on fixing that: [Homebrew/brew#13403](https://github.com/Homebrew/brew/pull/13403)
> 
> > is "formulae" a typo as the letter e is close to s?
> 
> Both "formulae" and "formulas" are recognised plural forms of "formula". We generally use the former.

what a relieve. thanks for the info.

Have you guys an opinion on the idea/feature I mentioned above? 
... "print the 50 (any given number) latest packages added" or "print all new packages added since 'date'|'days ago'"

Would you say it's worth a feature request? Or does this already exist?

**Reply by alebcay** [reply]
There's no command to simply do what you're asking here, but Homebrew taps are Git repositories under the hood, and you can query those however you may wish.

Print, for instance, the last 50 latest packages added (adjust `-m50` to a desired number):
```bash
git -C $(brew --repository)/Library/Taps/homebrew/homebrew-core log --pretty=format:%s | grep -m50 "(new formula)"
```

Print all new packages added in the past two weeks (adjust `2 weeks ago` to a desired date or time frame):
```bash
git -C $(brew --repository)/Library/Taps/homebrew/homebrew-core log --pretty=format:%s --since="2 weeks ago" | grep "(new formula)"
```

**Reply by marcotrosi** [reply]
> There's no command to simply do what you're asking here, but Homebrew taps are Git repositories under the hood, and you can query those however you may wish.
> 
> Print, for instance, the last 50 latest packages added (adjust `-m50` to a desired number):
> 
> ```shell
> git -C $(brew --repository)/Library/Taps/homebrew/homebrew-core log --pretty=format:%s | grep -m50 "(new formula)"
> ```
> 
> Print all new packages added in the past two weeks (adjust `2 weeks ago` to a desired date or time frame):
> 
> ```shell
> git -C $(brew --repository)/Library/Taps/homebrew/homebrew-core log --pretty=format:%s --since="2 weeks ago" | grep "(new formula)"
> ```

thanks @alebcay for the code, works great. ❤️

if I read it correctly, then this means that the information in parentheses "(new formula)" that grep is filtering for, is only available as long as there was no update available, because then this information would change to (update). So theoretically it's possible to miss new packages.

But if I'm the only one who is interest in discovering new packages then I think the commands you provided are more than enough for me. Thanks again, and have a great day. 😊

marco

**Reply by SMillerDev** [reply]
No, it's the commit message for new formula. Those won't change once a formula is updated.

**Reply by alebcay** [reply]
To clarify, this filters on new formulae that are added to Homebrew (i.e. previously had no version in our repo, and now are in our repo). It does not filter on formulae/packages that received an _update_ in our repo. Maybe I misunderstood what you're asking for.

Generally, you can omit the `grep` piece if you want to see updates too. But note that there can be other commits mixed in too if you do so.
